Update docs/configuring-exim-relay.md: on DKIM

luixxiul opened the pull request:
This PR intends to clarify how to enable DKIM support  for exim-relay with this role. DKIM is an authentication method which works consistently, and enabling it will greatly improve deliverability of emails sent with exim-relay. This should benefit playbooks which implement this role.

Commit:
Update docs/configuring-exim-relay.md: on DKIM

Signed-off-by: Suguru Hirahara <acioustick@noreply.codeberg.org>

spantaleev commented on docs/configuring-exim-relay.md:
```diff
@@ -41,9 +41,33 @@ exim_relay_sender_address: "example@{{ exim_relay_hostname }}"
+
+To enable DKIM support, at first you need to create a DKIM key pair. You can use DKIM with RSA signatures and Ed25519 elliptic curve signatures. For details about setting Ed25519 signatures (its cheacteristics and how to create Ed25519 key pair, etc.), you can refer an article such as [this one](https://www.mailhardener.com/kb/how-to-use-dkim-with-ed25519).
+
+After creating a key pair, add its **public key** to your domain's DKIM DNS record. Look on the internet for a guide about how to do so. Note that every DKIM record must have a unique identifier called as "selector". If you set `eximrelay` as a selector, your DKIM record's DNS name would be `eximrelay._domainkey.example.com`.
```
The DKIM selector is mentioned, but it's not mentioned:

- what the the default value (`default`) is, so people can configure DNS correctly
- how to change the default DKIM selector (perhaps just mentioning the new variable is enough)

Or do you intend to make this improvement in another commit later?

luixxiul replied:
I'll try to address it with this PR testing with PR testing it for my instance!

Commit:
Merge remote-tracking branch 'upstream/main' into dkim

Commit:
Update docs/configuring-exim-relay.md: mention the new variable exim_relay_dkim_selector

- Move the variable on defaults/main.yml above exim_relay_dkim_privkey_path to group the variables related to DKIM

Signed-off-by: Suguru Hirahara <acioustick@noreply.codeberg.org>

Commit:
Update docs/configuring-exim-relay.md: add "mash-exim-relay" to the troubleshooting section

Signed-off-by: Suguru Hirahara <acioustick@noreply.codeberg.org>

Commit:
Update docs/configuring-exim-relay.md: itelate

Signed-off-by: Suguru Hirahara <acioustick@noreply.codeberg.org>

Commit:
Update docs/configuring-exim-relay.md: replace an example

Signed-off-by: Suguru Hirahara <acioustick@noreply.codeberg.org>

luixxiul commented:
@spantaleev this is ready for review!

Commit:
Update docs/configuring-exim-relay.md: minor edit

Signed-off-by: Suguru Hirahara <acioustick@noreply.codeberg.org>
